design/lvm/mountable_snapshot.md

```diff
@@ -0,0 +1,191 @@
+---
+title: Snapshot Mount Support for LVM-LocalPV
+authors:
+  - "@wowditi"
+owners:
+  - "@kmova"
+creation-date: 2022-03-31
+last-updated: 2022-03-31
+status: Request for Comment
+---
+
+# Snapshot Mount Support for LVM-LocalPV
+
+## Table of Contents
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+* [Implementation Details](#implementation-details)
+* [Test Plan](#test-plan)
+* [Graduation Criteria](#graduation-criteria)
+* [Drawbacks](#drawbacks)
+
+## Summary
+
+LVM Snapshots are space efficient and quick point in time copies of lvm volumes. It consumes the space only when changes are made to the source logical volume. For testing purposes it can be beneficial to start a new application, for example a database, using such a snapshot as the backing storage volume. This allows for creating an exact replica of the application at the moment the snapshot was taken (assuming the state is entirely depedent on the filesystem), which can used to debug issues that occured in production or test how a new version of the application (or an external application) would affect the data.
+
+## Motivation
+
+### Goals
+
+- user should be able to mount snapshots
+- user should be able to mount thick and thin snapshots
+- user should be able to mount snapshots as read only and copy on write 
+
+### Non-Goals
+
+- Creating clones from Snapshots
+- restore of a snapshot
+
+
+## Proposal
+
+To mount a k8s lvmpv snapshot we need to create a persistent volume claim that references the snapshot as datasource:
+
+```
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: csi-lvmpvc-snap
+spec:
+  storageClassName: openebs-lvmpv
```

Comment:
Is storage class required to be of `local.csi.openebs.io` provisioner type? Can we have different storage class consuming the lvm snapshots?

Reply:
Since the `dataSource` is provided separately I don't think it's required. I'm not entirely sure what's involved with creating a new provisioner type, but it would be a good solution to the downside that I mentioned in the design.